### ISSUE TITLE
CalDAV Sync fails

### ISSUE THREAD

**Issue by Pieren** [marcoancona/MineTime#94]
It seems that my nextcloud CalDAV calendar doesn't sync properly every time. Even when I start it manually, the events are not synced.
The logs show up as follows:

```
2018-11-21T08:01:21.381Z - error: CalDAV: error while running syncCalendar -  message=Bad status on propstat: HTTP/1.1 418 I'm a teapot, stack=Error: Bad status on propstat: HTTP/1.1 418 I'm a teapot
    at getProp (/opt/MineTime/resources/app.asar/node_modules/dav/build/request.js:212:11)
    at Array.map (<anonymous>)
    at getProps (/opt/MineTime/resources/app.asar/node_modules/dav/build/request.js:229:31)
    at /opt/MineTime/resources/app.asar/node_modules/dav/build/request.js:177:39
    at Array.map (<anonymous>)
    at transformResponse (/opt/MineTime/resources/app.asar/node_modules/dav/build/request.js:176:37)
    at Basic.callee$2$0$ (/opt/MineTime/resources/app.asar/node_modules/dav/build/transport.js:104:44)
    at tryCatch (/opt/MineTime/resources/app.asar/node_modules/regenerator-runtime/runtime.js:62:40)
    at Generator.invoke [as _invoke] (/opt/MineTime/resources/app.asar/node_modules/regenerator-runtime/runtime.js:296:22)
    at Generator.prototype.(anonymous function) [as next] (/opt/MineTime/resources/app.asar/node_modules/regenerator-runtime/runtime.js:114:21)
    at onFulfilled (/opt/MineTime/resources/app.asar/node_modules/co/index.js:65:19)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
2018-11-21T08:01:21.384Z - error: Error on sync of calendar 5bf26e71f1e6be18f31bbcff. Next trial in in 30 secs... message=Bad status on propstat: HTTP/1.1 418 I'm a teapot, stack=Error: Bad status on propstat: HTTP/1.1 418 I'm a teapot
    at getProp (/opt/MineTime/resources/app.asar/node_modules/dav/build/request.js:212:11)
    at Array.map (<anonymous>)
    at getProps (/opt/MineTime/resources/app.asar/node_modules/dav/build/request.js:229:31)
    at /opt/MineTime/resources/app.asar/node_modules/dav/build/request.js:177:39
    at Array.map (<anonymous>)
    at transformResponse (/opt/MineTime/resources/app.asar/node_modules/dav/build/request.js:176:37)
    at Basic.callee$2$0$ (/opt/MineTime/resources/app.asar/node_modules/dav/build/transport.js:104:44)
    at tryCatch (/opt/MineTime/resources/app.asar/node_modules/regenerator-runtime/runtime.js:62:40)
    at Generator.invoke [as _invoke] (/opt/MineTime/resources/app.asar/node_modules/regenerator-runtime/runtime.js:296:22)
    at Generator.prototype.(anonymous function) [as next] (/opt/MineTime/resources/app.asar/node_modules/regenerator-runtime/runtime.js:114:21)
    at onFulfilled (/opt/MineTime/resources/app.asar/node_modules/co/index.js:65:19)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7), timestamp=2018-11-21T08:01:21.381Z
```

**Comment by marcoancona** [comment]
Is this a self-hosted nextcloud or provided by a third party?

**Comment by Pieren** [comment]
It's self-hosted on a server provided by a third party.

**Comment by Pieren** [comment]
As you added the "need more info" label: What information do you need from me? :)
This bug is still forcing me to relink my calendars like every day.

**Comment by marcoancona** [comment]
Unfortunately, it is very difficult to track the problem if we cannot reproduce it. Would it be possible to create a test account for us with the same provider?
Other information that might help:
- when does the problem occur (app start, after a while the app is started, upon event editing, triggering a manual sync...)?
- you mentioned that relinking help: do you also reset the calendar data or remove/add the account? After relinking sync works again, if so, for how long?

**Comment by marcoancona** [comment]
Several changes have been made to CalDAV sync since the opening of this issue. I will close as we have no additional information. Feel free to comment if the problem persists.

**Comment by marcoancona** [comment]
Fixed in 1.8.4.
